tests/components/tts/test_media_source.py

```diff
@@ -195,3 +200,66 @@ async def test_resolving_errors(hass: HomeAssistant, setup: str) -> None:
             "media-source://tts/non-existing?message=bla&non_existing_option=bla",
             None,
         )
+
+
+@pytest.mark.parametrize(
+    ("setup", "result_engine"),
+    [
+        ("mock_setup", "test"),
+        ("mock_config_entry_setup", "tts.test"),
+    ],
+    indirect=["setup"],
+)
+async def test_generate_media_source_id_and_media_source_id_to_kwargs(
```

Comment:
There's some existing coverage of the `generate_media_source_id` method in the `test_init.py` file (https://github.com/home-assistant/core/blob/dev/tests/components/tts/test_init.py#L1320). The initial JSON encoding part of this PR could be tested there to keep it in one place.

Reply:
I personally prefer having the tests in the file that matches the code under test. In a separate PR I can move that test here. I also prefer having smaller targeted tests. The test here tests that when we go from args to URL and then back to args we haven't lost any information.